lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -2556,3 +2672,119 @@ void mme_app_handle_modify_ue_ambr_request(
   unlock_ue_contexts(ue_context_p);
   OAILOG_FUNC_OUT(LOG_MME_APP);
 }
+
+/**
+ * This Function handles PCRF initiated
+ * Dedicated bearer activation Request message from SGW
+ */
+void mme_app_handle_nw_init_ded_bearer_actv_req(
+  const itti_s11_nw_init_actv_bearer_request_t
+  *const nw_init_bearer_actv_req_p)
+{
+  ue_mm_context_t *ue_context_p = NULL;
+  OAILOG_FUNC_IN(LOG_MME_APP);
+
+  OAILOG_INFO(
+    LOG_MME_APP,
+    "Received Dedicated bearer activation Request from SGW with LBI %d\n",
+    nw_init_bearer_actv_req_p->lbi);
+
+  ebi_t linked_eps_bearer_id = nw_init_bearer_actv_req_p->lbi;
+  ue_context_p = mme_ue_context_exists_s11_teid(
+    &mme_app_desc.mme_ue_contexts, nw_init_bearer_actv_req_p->s11_mme_teid);
+
+  if (ue_context_p == NULL) {
+    MSC_LOG_RX_DISCARDED_MESSAGE(
+      MSC_MMEAPP_MME,
+      MSC_S11_MME,
+      NULL,
+      0,
+      "0 PCRF initiated Bearer Actv Req local S11 teid " TEID_FMT " ",
+      mme_teid_S11->teid);
+
+    OAILOG_ERROR(
+      LOG_MME_APP,
+      "We didn't find this teid in list of UE: %08x\n",
+      nw_init_bearer_actv_req_p->s11_mme_teid);
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+
+  bearer_context_t *linked_bc =
+    mme_app_get_bearer_context(ue_context_p, linked_eps_bearer_id);
+  if (!linked_bc) {
+    MSC_LOG_RX_DISCARDED_MESSAGE(
+      MSC_MMEAPP_MME,
+      MSC_S11_MME,
+      NULL,
+      0,
+      "0 CREATE_BEARERS_REQUEST ue id " MME_UE_S1AP_ID_FMT
+      " local S11 teid " TEID_FMT " ",
+      ue_context_p->mme_ue_s1ap_id,
+      nw_init_bearer_actv_req_p->teid);
+    OAILOG_ERROR(
+      LOG_MME_APP,
+      "We didn't find the linked bearer id %" PRIu8
+      " for UE: " MME_UE_S1AP_ID_FMT "\n",
+      linked_eps_bearer_id,
+      ue_context_p->mme_ue_s1ap_id);
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+  pdn_cid_t cid = linked_bc->pdn_cx_id;
+  // forward request to NAS
+  MessageDef *message_p =
+    itti_alloc_new_message(TASK_MME_APP, MME_APP_CREATE_DEDICATED_BEARER_REQ);
+  if (message_p == NULL) {
+    OAILOG_ERROR(
+    LOG_MME_APP,
+    "itti_alloc_new_message failed for MME_APP_CREATE_DEDICATED_BEARER_REQ\n");
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+  MME_APP_CREATE_DEDICATED_BEARER_REQ(message_p).ue_id =
+    ue_context_p->mme_ue_s1ap_id;
+  MME_APP_CREATE_DEDICATED_BEARER_REQ(message_p).cid = cid;
+  //EBI Will be assigned by NAS Task
+  MME_APP_CREATE_DEDICATED_BEARER_REQ(message_p).ebi = 0;
+  MME_APP_CREATE_DEDICATED_BEARER_REQ(message_p).linked_ebi =
+    ue_context_p->pdn_contexts[cid]->default_ebi;
+  MME_APP_CREATE_DEDICATED_BEARER_REQ(message_p).bearer_qos =
+    nw_init_bearer_actv_req_p->eps_bearer_qos;
+  MME_APP_CREATE_DEDICATED_BEARER_REQ(message_p).gtp_teid =
```

Comment:
Why are we sending TE Id it NAS ? Store in MME and use it later while sending message to eNB.

Reply:
Bearer context for the new dedicated bearer gets created at NAS similar to default bearer context. Hence gtp_teid is sent from MME to NAS. Once the bearer context is created, gtp_teid is stored in the context by NAS and MME fetches it from the context to send it in ERAB SETUP REQ tp eNB

Reply:
Sending gtp_teid is not a good design. NAS code shall not have any knowledge of tunnel ids . We can discuss this in next call.